### PR TITLE
feat: [MH-000] fetch module

### PR DESCRIPTION
**Pull request by EvgeniyKvasyuk**
#### Description

Simple abstraction above "axios", which give possibility to set fetcher with particular settings with interceptors and configs

| Details                     | Description                                      |
| --------------------------- | ------------------------------------------------ |
| JIRA ticket                 |  |
| Design source               | N/A                                              |
| Related PRs                 | N/A                                              |
| Documentation update needed | No                                               |
| QA sign-off by              | N/A                                              |

## Checklist

- [x] [Commit message convention](https://kb.epam.com/display/MAHE/Code+Style) followed.
- [x] [Branch naming convention](https://kb.epam.com/display/MAHE/Code+Style) followed.
- [x] Performed self review & removed the debugging code (debugger, console log etc).
- [x] Added explanatory comment for tricky code explaining "why".
- [x] Tested on major browsers (Chrome, Firefox, Safari, Edge)
- [x] Manual testing or integration testing done.

### CODE REVIEW

File: src/common/fetch/Fetch.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    return {
+      response: responsePromise,
+      abort: () => {
+        controller.abort();
+      },
+    };
+  }
+
+  get(request: Omit<ExtendedAxiosRequestConfig, 'method'>) {
+    return this.request({ ...request, method: 'GET' });
```
Comment (VladSolodjankin) [inline src/common/fetch/Fetch.ts:59]: I think we can use constants METHODS for method: 'GET'

File: src/common/fetch/constants.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+export const METHODS = {
```
Comment (VladSolodjankin) [inline src/common/fetch/constants.ts:1]: maybe it is better to use Enum instead of const?

File: src/common/fetch/Fetch.ts (hunk 1 above)
Comment (EvgeniyKvasyuk) [reply src/common/fetch/Fetch.ts:59]: Y’r absolutely right

File: src/common/fetch/constants.ts (hunk 2 above)
Comment (EvgeniyKvasyuk) [reply src/common/fetch/constants.ts:1]: Tried to use enums, but this method parameter  is forwarded to axios it have to be string:(

File: src/common/fetch/Fetch.ts [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    this.axios.interceptors.response.use(onSuccess as ResponseSuccessCallback, onError as RejectedCallback);
+  }
+
+  request({ onSuccess, onError, ...request }: ExtendedAxiosRequestConfig) {
+    const controller = new AbortController();
+
+    const responsePromise = this.axios
+      .request({ ...request, signal: controller.signal } as ExtendedAxiosRequestConfig)
+      .then((data) => {
+        // простой способ обработки запроса через коллбеки + сохраняется возможность обрабатывать Promise и по месту вызова
```
[the remark this answers is not in the document]
Comment (EvgeniyKvasyuk) [reply src/common/fetch/Fetch.ts:43]: My bad:(

**Review by VladSolodjankin — APPROVED** [review]

**Pull request merged**